virtual-server/index.js

```diff
@@ -0,0 +1,110 @@
+var ENDPOINT = 'api/quotations';
+
+// Register the worker and show the list of quotations.
+if (navigator.serviceWorker.controller) {
+  loadQuotations();
+} else {
+  navigator.serviceWorker.oncontrollerchange = function() {
+    this.controller.onstatechange = function() {
+      if (this.state === 'activated') {
+        loadQuotations();
+      }
+    };
+  };
+  navigator.serviceWorker.register('service-worker.js');
+}
+
+// When clicking add button, get the new quote and author and post to
+// the backend.
+document.getElementById('add-form').onsubmit = function(event) {
+  // Avoid navigation
+  event.preventDefault();
+
+  var newQuote = document.getElementById('new-quote').value.trim();
+  // Skip if no quote provided.
+  if (!newQuote) { return; }
+
+  // Leave blank to represent an anonymous quote.
+  var quoteAuthor = document.getElementById('quote-author').value.trim() || 'Anonymous';
+  var quote = { text: newQuote, author: quoteAuthor };
+  var headers = { 'content-type': 'application/json' };
+
+  // Send the API request. In this case, a `POST` on `quotations` collection.
+  fetch(ENDPOINT, { method: 'POST', body: JSON.stringify(quote), headers: headers })
+    .then(function(response) {
+      return response.json();
+    })
+    .then(function(addedQuote) {
+      document.getElementById('quotations').appendChild(getRowFor(addedQuote));
```

Comment:
Before merging this PR I think we need to clear out the `quote-author` and `new-quote` inputs so that the user understand that something has happened.